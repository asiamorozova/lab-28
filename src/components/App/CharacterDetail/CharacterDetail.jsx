import React from 'react'; 
import PropTypes from 'prop-types';


const CharacterDetail = ({ name, photo, quests, appearances }) => (
  <section> 
    <h1>{name}</h1>
    <img src={photo} />
    <p>Quests: {quests}</p>
    <p>Appearances:{appearances}</p>


  </section>
);

CharacterDetail.propTypes = {
  name: PropTypes.string.isRequired,
  photo: PropTypes.string.isRequired,
  quests: PropTypes.string.isRequired, 
  appearances: PropTypes.string.isRequired


};

export default CharacterDetail;

//[{"name":"Flukeman",
//"gender":"Genderless (possibly formerly male)",
//"status":"Alive",
//"born":null,
//"occupation":null,
//"rank":null,
//"affiliations":null,
//"portrayedby":"Darin Morgan",
//"image":"https://vignette.wikia.nocookie.net/x-files/images/e/e6/Flukeman_The_X_Files.png/revision/latest/scale-to-width-down/310?cb=20160216194606",
//"description":"The Flukeman was a genetic mutant that found its way into the New Jersey sewer system and killed several people, before being washed up in Martha's Vineyard. (TXF: \"The Host\", \"Pusher\")",
//"categories":["Beings and creatures","Monster of the Week","Mutants"]}]
