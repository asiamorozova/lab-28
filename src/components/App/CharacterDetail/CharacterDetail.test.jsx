import React from 'react';
import { shallow } from 'enzyme';
import CharacterDetail from './CharacterDetail';

describe('Character Detail component', () => {
  it('renders Character Detail', () => {
    const wrapper = shallow(<CharacterDetail name= "Fox Maulder"
      image="maulder.png" description='he wants to believe' categories="every episode except that one season without him" />);
    expect(wrapper).toMatchSnapshot();
  });
});

