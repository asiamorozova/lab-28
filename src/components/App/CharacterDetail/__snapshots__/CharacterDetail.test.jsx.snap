// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Detail component renders Character Detail 1`] = `
<section>
  <h1>
    Fox Maulder
  </h1>
  <img />
  <p>
    Quests: 
  </p>
  <p>
    Appearances:
  </p>
</section>
`;
